cmd/kubeadm/app/phases/kubeconfig/kubeconfig.go

```diff
@@ -362,25 +338,36 @@ func ValidateKubeconfigsForExternalCA(outDir string, cfg *kubeadmapi.InitConfigu
 		kubeadmconstants.SchedulerKubeConfigFileName,
 	}
 
+	return handleKubeConfigFiles(validateKubeConfig, outDir, cfg, kubeConfigFileNames...)
+}
+
+// createKubeConfigFiles creates all the requested kubeconfig files.
+func handleKubeConfigFiles(fn handleKubeConfig, outDir string, cfg *kubeadmapi.InitConfiguration, kubeConfigFileNames ...string) error {
```

Comment:
I would rather have the full `func(outDir, filename string, config *clientcmdapi.Config) error` here, than this type.
It's easier to see what the func takes, plus we don't have another type, that's used just here and nowhere else.

Reply:
@ping035627 
please take a look at @rosti 's comment here. don't define a new type `handleKubeConfig`.

but my preference instead of passing function objects around would be to call break the logic into separate utility functions, even if we have to iterate over list of configs twice. the lists are small...

Reply:
@ping035627 ^^